optuna/terminator/improvement/evaluator.py

```diff
@@ -128,3 +128,49 @@ def _validate_input(
                 "The intersection search space is empty. This condition is not supported by "
                 f"{cls.__name__}."
             )
+
+
+@experimental_class("3.4.0")
+class BestValueStagnationEvaluator(BaseImprovementEvaluator):
+    """Evaluates the stagnation period of the best value in an optimization process.
+
+    This class is initialized with a maximum stagnation period (`max_stagnation_trials`)
+    and is designed toevaluate the remaining trials before reaching this maximum period
+    of allowed stagnation.
+
+    Args:
+        max_stagnation_trials (int):
```

Comment:
Thank you for your update. The type information will be added by sphinx, so could you remove it, please?
```suggestion
        max_stagnation_trials:
```